src/frontends/src/json_util.hpp

```diff
@@ -0,0 +1,203 @@
+#include <rapidjson/document.h>
+#include <clipper/datatypes.hpp>
+#include <stdexcept>
+
+using clipper::Input;
+using clipper::InputType;
+using clipper::Output;
+using clipper::OutputType;
+using clipper::VersionedModelId;
+using rapidjson::Type;
+
+static std::vector<std::string> kTypeNames = {
+    "Null", "False", "True", "Object", "Array", "String", "Number"};
+
+// RapidJSON value types
+/*
+enum Type {
+  kNullType = 0,      // null
+  kFalseType = 1,     // false
+  kTrueType = 2,      // true
+  kObjectType = 3,    // object
+  kArrayType = 4,     // array
+  kStringType = 5,    // string
+  kNumberType = 6     // number
+}; */
+
+namespace clipper_json {
+
+class json_parse_error : public std::runtime_error {
+ public:
+  json_parse_error(const std::string& what) : std::runtime_error(what) {}
+  ~json_parse_error() throw(){};
+};
+
+class json_semantic_error : public std::runtime_error {
```

Comment:
Do you mean something like
```
<error message>
Expected JSON format for feedback query request:
   * {
   *  "uid" := string,
   *  "input" := [double] | [int] | [string] | [byte] | [float],
   *  "model_name" := string,
   *  "model_version" := int,
   *  "label" := double
   * }
```? Might be a little too much and also not completely appropriate for `json_semantic_error` in the current form (especially since the exception gets thrown pretty far down at the elementary parsing level). I think that at the very least, we can send the user an HTTP response containing a basic error message (from the exception) and the expected schema (by changing the arguments to the `respond_http` calls in the respective RequestHandler implementations). In the future, it might be worth trying to parse each argument rather than just stopping at the first error so that we can enumerate out each issue with the input JSON, but we may have to account for increased overhead in trying to compile these more useful error messages.